share/availability/discovery/discovery.go

```diff
@@ -122,6 +122,30 @@ func (d *Discovery) Peers(ctx context.Context) ([]peer.ID, error) {
 	return d.set.Peers(ctx)
 }
 
+// Discard removes the peer from the peer set and rediscovers more if soft peer limit is not
+// reached. Reports whether peer was removed with bool.
+func (d *Discovery) Discard(id peer.ID) bool {
+	if !d.set.Contains(id) {
+		return false
+	}
+
+	d.host.ConnManager().Unprotect(id, rendezvousPoint)
+	d.connector.Backoff(id)
+	d.set.Remove(id)
+	d.onUpdatedPeers(id, false)
+	log.Debugw("removed peer from the peer set", "peer", id)
+
+	if d.set.Size() < d.set.Limit() {
+		// trigger discovery
+		select {
+		case d.triggerDisc <- struct{}{}:
+		default:
+		}
+	}
```

Comment:
could make a method to DRY this but nit

Reply:
I would, but it is only used in one place.